compiler/rustc_type_ir/src/index.rs

```diff
@@ -0,0 +1,76 @@
+#[derive(Copy, Clone, Hash, Eq, PartialEq, Ord, PartialOrd, Debug)]
```

Comment:
re: this and the `index_nightly` module -- I wonder if I should instead just pull out the `newtype_index` macro itself into a separate proc-macro crate that depends on fewer nightly features than rustc_macros. Then both nightly/stable can share the macro definitions rather than having this sucky duplicated one.